File: SCC0600/exerciciosC/ex45.c
#include<stdio.h>

int main (void) {
    int size;
    scanf("%d", &size);
    int arr[size];
    for (int i = 0; i < size; i++) {
        scanf("%d", &arr[i]);
    }
    int i, j;
    scanf("%d", &i);
    scanf("%d", &j);
    int maior = i < j ? j : i;
    int menor = i < j ? i : j;
    if (menor < 0 || maior >= size) {
        printf("%s", "INVALIDO");
        return -1;
    }
    int soma = 0;
    for (int k = menor; k <= maior; k++) {
        soma += arr[k];
    }
    printf("%d", soma);
}

File: SCC0600/exercíciosPy/ex45.py
n1 = int(input())
v1 = str(input())
nums = str(input())

def numerize (arr, rang):
  newarr = [0] * rang
  splited = arr.split(' ')
  for k in range(rang):
    newarr[k] = int(splited[k])
  return newarr

newv1 = numerize(v1, n1)
newnums = numerize(nums, 2)
newnums.sort()

soma = 0
if abs(newnums[0]) < n1 and abs(newnums[1]) < n1:
  for i in range(newnums[0], newnums[1] + 1):
    soma += newv1[i]
else:
  soma = 'INVALIDO'

print(soma)